client/state/products-list/selectors/compute-full-and-monthly-prices-for-plan.ts

```diff
@@ -66,21 +99,21 @@ function computePricesForWpComPlan( state, siteId, planObject ) {
 }
 
 /**
- * Compute the Jetpack Search product tiered price based on the site's number of posts.
- *
- * @param {object} state Current redux state
- * @param {number} siteId Site ID to consider
- * @param {object} productSlug The product slug
- * @returns {number} The tiered price (given the site's number of published posts)
+ * Compute the Jetpack Search product tiered price based on the site's number of published posts.
  */
-function getSearchProductTierPrice( state, siteId, productSlug ) {
+function getSearchProductTierPrice(
+	state: AppState,
+	siteId: number,
+	productSlug: string
+): number | null {
 	const postsCounts = getAllPostCounts( state, siteId, 'post' );
 	const postsCount = postsCounts?.publish || 0;
 	const priceTierList = getProductPriceTierList( state, productSlug );
-	const tier = priceTierList.filter(
-		( tierItem ) => postsCount >= tierItem.minimum_units && postsCount <= tierItem.maximum_units
-	);
-	const minPrice = tier[ 0 ].minimum_price.toString(); // is missing decimal, ie- $9.95 is returned as 995
```

Comment:
There's always the possibility that no tier will be found, which would have caused this line to throw a fatal error. In fact, such a recorded fatal error was exactly what inspired this PR.